src/table.ts

```diff
@@ -81,4 +148,79 @@ export class Table {
 
     return sql.join(" ") + ";";
   }
+
+  addField(field: ColumnDefinition) {
+    this.columns.push(field);
+    return this;
+  }
+
+  /**
+   * Execute query to create table.
+   * 
+   * @param adapter Custom database adapter
+   */
+  public async execute(
+    options?: { adapter?: Adapter } & QueryOptions,
+  ): Promise<QueryResult<any>> {
+    let currentAdapter = options?.adapter || this.adapter;
+
+    if (!currentAdapter) {
+      throw new Error("Cannot run query, adapter is not provided!");
+    }
+
+    return await currentAdapter.query(this.toSQL(), options);
+  }
+}
+
+export class TableInfo {
+  private options: Required<TableOptions>;
+
+  constructor(
+    /** Table name on the database */
+    private tableName: string,
+    /** Adapter */
+    private adapter?: Adapter,
+    /** Other options */
+    options?: TableOptions,
+  ) {
+    // Populate `this.options` with default
+    this.options = Object.assign({}, { createIfNotExists: false }, options);
+  }
+
+  /** Checks if this Table exists. */
+  async exists(): Promise<boolean> {
+    if (!this.adapter) {
+      throw new Error("Cannot run query, adapter is not provided!");
+    }
+    // SELECT name FROM sqlite_master WHERE type='table' AND name='${tablename}';
+
+    switch (this.adapter.type) {
```

Comment:
Even though this code works, it's pretty hard to read. You need to refactor it into something like this:

```ts
switch (this.adapter.type) {
  case "sqlite":
    const sqliteQuery = await this.adapter.query(
      `SELECT name FROM sqlite_master WHERE type='table' AND name='${this.tableName}'`,
    );
    return sqliteQuery.records.length === 1;
  case "postgres":
    const postgresQuery = await this.adapter.query(
      `SELECT EXISTS (SELECT FROM pg_tables WHERE tablename = '${this.tableName}')`,
    );
    if (postgresQuery.records[0]) {
      let result = postgresQuery.records[0] as any;
      return result.exists;
    } else {
      return false;
    }
  case "mysql":
    const mysqlQuery = await this.adapter.query(
      `SELECT * FROM information_schema.tables WHERE table_name = '${this.tableName}' LIMIT 1`,
    );
    return mysqlQuery.records.length === 1;
  default:
    return false;
}
```

This code works exactly the same as yours.

Reply:
Done!